specification/threatmodel.adoc

```diff
@@ -1,187 +1,300 @@
-| Operational Features | QOS interoperability | Implementation-specific | Interoperability with QoS features for TVM workload | Malicious host uses QoS capabilities as a side-channel | QOS SIG to specify 
-| Operational Features | RAS interoperability | Implementation-specific | Interoperability with RAS features for TVM workload | Malicious host uses RAS capabilities as a side-channel or to cause integrity violations | RAS SIG to specify 
+| I/O Protection | Trusted I/O from devices admitted into the TCB of a TVM |
+Implementation-specific | Device attestation, Link protection, IOMMU |
+Admission control to bind devices to TEEs | CoVE-IO, IOMMU
+
+| Secure IRQ | Trusted Interrupts | Required | Secure interrupt files, MMU, MPU,
+ MTT | Prevent IRQ injections that violate priority or masking | Smsdia w/ AIA
+
+| Secure Timetamp | Trusted timestamps | Required | Confidential supervisor
+domain qualifier for CSR accesses | Ensure TEE have consistent timestamp view |
+CoVE ABI
+
+| Debug & Profile | Trusted performance monitoring unit data | Required |
+Confidential supervisor domain context switch of perf. mon. counters  |
+Ensure TEEs get correct PMU info; prevent data leakage due to PMU information
+(fingerprint attacks) | CoVE ABI
+
+| Debug & Profile | Secure External Debug support | Required | Confidential
+supervisor domain qualifier for External debug controls | Support debug trigger
+registers for TVM | Secure Debug, CoVE ABI
+
+| Debug & Profile | Authenticated debug (Production device) | Required |
+Authorize debug via TEE RoT | Ensure hardware debug prob (e.g., JTAG SWD) is
+disabled in production | Secure Debug, CoVE ABI
+
+| Availability | TVM DoS Protection | Required | VMM retains ability to
+interrupt TVM | Prevent TVM from refusing to exit | CoVE ABI, Smsdia
+
+| Availability | VMM DoS Protection | Implementation-specific | Not in scope
+for CoVE | Prevent non-TCB components from refusing to run TEE | Not applicable
+
+| Side Channel | Protected address mapping (controlled side channel) | Required
+| Confidential mode qualifier, cryptography, MMU/MPU, MTT | Similar to memory
+remapping attacks | CoVE ABI, Security Model
+
+| Side Channel | Micro-architectural side channels (branch prediction | Required
+| uArch state flushing, entropy defenses | Prevent attacks such as meltdown or
+spectre (it is difficult to defend against such attacks in advance) | Security
+Model
+
+| Side Channel | Control channels, single-step/zero-step attacks | Required |
+uArch state flushing, entropy defenses | Prevent interrupt/exception injection
+(combined with cache side channel to leak sensitive data) | Security Model
+
+| Side Channel | Architectural cache side channel | Implementation-specific  |
+uArch state flushing, entropy defenses | Prevent shared resource contention,
+e.g. attacks such as prime probe | Security Model
+
```

Comment:
Not clear what/how uarch state flushing or entropy defenses help against cache side channels.

Reply:
yeah this was a copy/paste error - will mention cache partitioning as an example strategy for this case.